app/src/main/java/com/vci/vectorcamapp/imaging/presentation/ImagingScreen.kt

```diff
@@ -47,54 +50,72 @@ fun ImagingScreen(
-                    modifier = modifier,
-                    specimenBitmap = specimenBitmap,
-                    onSpecimenIdCorrected = { onAction(ImagingAction.CorrectSpecimenId(it)) },
-                    onRetakeImage = { onAction(ImagingAction.RetakeImage) },
-                    onSaveImageToSession = { onAction(ImagingAction.SaveImageToSession) }
-                )
-            }
+                state.currentImageBytes != null -> {
+                    val specimenBitmap = remember(state.currentImageBytes) {
+                        BitmapFactory.decodeByteArray(
+                            state.currentImageBytes,
+                            0,
+                            state.currentImageBytes.size
+                        )
+                    }
+
+                    CapturedSpecimenOverlay(
+                        specimen = state.currentSpecimen,
+                        inferenceResult = state.currentInferenceResult,
+                        modifier = Modifier,
+                        specimenBitmap = specimenBitmap,
+                        onSpecimenIdCorrected = { onAction(ImagingAction.CorrectSpecimenId(it)) },
+                        onRetakeImage = { onAction(ImagingAction.RetakeImage) },
+                        onSaveImageToSession = { onAction(ImagingAction.SaveImageToSession) }
+                    )
+                }
 
-            else -> {
-                LiveCameraPreviewPage(
-                    controller = controller,
-                    inferenceResults = state.previewInferenceResults,
-                    onImageCaptured = {
-                        onAction(ImagingAction.CaptureImage(controller))
-                    },
-                    onSaveSessionProgress = { onAction(ImagingAction.SaveSessionProgress) },
-                    onSubmitSession = { onAction(ImagingAction.SubmitSession) },
-                    manualFocusPoint = state.manualFocusPoint,
-                    onAction = onAction,
-                    modifier = modifier,
-                    captureEnabled = !state.isProcessing
-                )
+                else -> {
+                    LiveCameraPreviewPage(
+                        controller = controller,
+                        inferenceResults = state.previewInferenceResults,
+                        onImageCaptured = {
+                            onAction(ImagingAction.CaptureImage(controller))
+                        },
+                        onSaveSessionProgress = { onAction(ImagingAction.SaveSessionProgress) },
+                        onSubmitSession = { onAction(ImagingAction.SubmitSession) },
+                        manualFocusPoint = state.manualFocusPoint,
+                        onAction = onAction,
+                        modifier = Modifier,
+                        captureEnabled = !state.isProcessing
+                    )
+                }
             }
         }
+
+        if (state.isLoading) {
```

Comment:
I'd say move this to the navGraph. Because the LoadingAnimation is not a property of the imaging screen.